### PR TITLE
Task context + schema

### LINKED ISSUE #815: Backup and Restore tasks show on Azure DB Manage page
- SQL Operations Studio Version: `master`

Steps to Reproduce:
1. Connect to an Azure SQL DB and open Database Manage page

Expected: No tasks that won't work are visible

Actual: Backup and Restore tasks buttons are visible

### PR DESCRIPTION
**Pull request by anthonydresser**
Updates the task schema and config to make a little more since.

Now the task widget takes either null, or an array of either strings or an object that has a name field and a when field, where the when field is a way to specify the context to show the task.
fixes #815

### CODE REVIEW

File: build/lib/i18n.js [hunk 1]
```diff
@@ -533,12 +531,9 @@ function getResource(sourceFile) {
     else if (/^vs\/workbench/.test(sourceFile)) {
         return { name: 'vs/workbench', project: workbenchProject };
     }
-
-    // {{SQL CARBON EDIT}}
```
Comment (AbbiePetcht) [inline build/lib/i18n.js]: >// {{SQL CARBON EDIT}} [](start = 4, length = 22)

revert this file

**Comment by AbbiePetcht** [comment]
@anthonydresser  can you link the issues that you are fixing?

File: src/sql/platform/dashboard/common/widgetRegistry.ts [hunk 2]
```diff
@@ -6,8 +6,10 @@ import { IInsightsConfig } from 'sql/parts/dashboard/widgets/insights/interfaces
 
 import * as platform from 'vs/platform/registry/common/platform';
 import { IJSONSchema, IJSONSchemaMap } from 'vs/base/common/jsonSchema';
+import { IJSONContributionRegistry, Extensions as JSONExtensions } from 'vs/platform/jsonschemas/common/jsonContributionRegistry';
 import * as nls from 'vs/nls';
-import { deepClone } from 'vs/base/common/objects';
+
+const contributionRegistry = platform.Registry.as<IJSONContributionRegistry>(JSONExtensions.JSONContribution);
```
Comment (AbbiePetcht) [inline src/sql/platform/dashboard/common/widgetRegistry.ts]: >contributionRegistry [](start = 6, length = 20)

do you need this?

File: src/sql/parts/dashboard/pages/serverDashboardPage.contribution.ts [hunk 3]
```diff
@@ -77,7 +77,7 @@ let defaultVal = [
 	{
 		name: 'Tasks',
 		widget: {
-			'tasks-widget': {}
+			'tasks-widget': [{ name: 'backup', when: '!mssql:iscloud' }, 'configureDashboard', 'newQuery']
```
Comment (AbbiePetcht) [inline src/sql/parts/dashboard/pages/serverDashboardPage.contribution.ts]: >backup [](start = 29, length = 6)

isn't this suppose to be restore? I am not sure. Please verify with Karl or Eric

Comment (AbbiePetcht) [reply src/sql/parts/dashboard/pages/serverDashboardPage.contribution.ts]: SSMS shows restore at the server level

---
In reply to: [172697411](https://github.com/Microsoft/sqlopsstudio/pull/849#discussion_r172697411) [](ancestors = 172697411)

Comment (anthonydresser) [reply src/sql/parts/dashboard/pages/serverDashboardPage.contribution.ts]: @erickangMSFT can you confirm? what pages are restore and backup supposed to appear on.

**Comment by AbbiePetcht** [comment]
fix #815: Backup and Restore tasks show on Azure DB Manage page

File: src/sql/parts/dashboard/pages/serverDashboardPage.contribution.ts (hunk 3 above)
Comment (erickang7) [reply src/sql/parts/dashboard/pages/serverDashboardPage.contribution.ts]: on-prem:
- Server Dashboard: Restore and Backup
- Database Dashboard: Restore and Backup

Azure
- Server Dashboard: none
- Database Dashboard: Restore in the future with Restore dialog support for Azure. Until then none.

**Pull request merged**